Patch based on upstream commit 066aeb918db7504f21ee9127b35603a06101a72a

The PyPdf2 package changed a lot and was renamed to just `pypdf`.
---

diff --git a/docs/Installation.rst b/docs/Installation.rst
index ec55082..ab18154 100644
--- a/docs/Installation.rst
+++ b/docs/Installation.rst
@@ -51,11 +51,11 @@ If this is not provided, please refer to the
 `pip homepage <https://pip.pypa.io/en/stable/installing/>`_ for help.
 
 
-Optionally you might want to install `PyPDF2`
+Optionally you might want to install `pypdf`
 - which is a requirement for |pdfposter| -
 provided by your distribution or vendor
 so at least this package will be maintained by your distribution.
-Check for a package named ``python-pypdf2`` or that like.
+Check for a package named ``python3-pypdf`` or that like.
 
 Then continue with :ref:`installing pdfposter` below.
 
@@ -65,7 +65,7 @@ Then continue with :ref:`installing pdfposter` below.
 Installing |pdfposter| using :command:`pip`
 ---------------------------------------------
 
-After installing `Python` (and optionally `PyPDF2`), just run::
+After installing `Python` (and optionally `pypdf`), just run::
 
   sudo pip install pdfposter
 
diff --git a/pdftools/pdfposter/__init__.py b/pdftools/pdfposter/__init__.py
index 838795e..6b4a36d 100644
--- a/pdftools/pdfposter/__init__.py
+++ b/pdftools/pdfposter/__init__.py
@@ -24,9 +24,9 @@ __copyright__ = "Copyright 2008-2022 by Hartmut Goebel <h.goebel@crazy-compilers
 __license__ = "SPDX-License-Identifier: GPL-3.0-or-later"
 __version__ = "0.8.1"
 
-from PyPDF2 import PdfWriter, PdfReader, PageObject
-from PyPDF2.types import NameObject
-from PyPDF2.generic import ContentStream, RectangleObject, IndirectObject
+from pypdf import PdfWriter, PdfReader, PageObject
+from pypdf.types import NameObject
+from pypdf.generic import ContentStream, RectangleObject, IndirectObject
 
 import logging
 from logging import log
diff --git a/pdftools/pdfposter/cmd.py b/pdftools/pdfposter/cmd.py
index 51d87aa..9a20357 100644
--- a/pdftools/pdfposter/cmd.py
+++ b/pdftools/pdfposter/cmd.py
@@ -27,7 +27,7 @@ __license__ = "SPDX-License-Identifier: GPL-3.0-or-later"
 from . import main, __version__, DEFAULT_MEDIASIZE, papersizes, DecryptionError
 from .i18n import _
 import re
-import PyPDF2.errors
+import pypdf.errors
 import argparse
 
 # pattern for parsing user textual box spec
@@ -192,7 +192,7 @@ try:
         main(args, infilename=args.infilename, outfilename=args.outfilename)
     except DecryptionError as e:
         raise SystemExit(str(e))
-    except PyPDF2.errors.PdfReadError as e:
+    except pypdf.errors.PdfReadError as e:
         parser.error('The input-file is either currupt or no PDF at all: %s'
                      % e)
 
diff --git a/setup.cfg b/setup.cfg
index 9616800..1e2f2d6 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -41,7 +41,7 @@ namespace_packages = pdftools
 include_package_data = True
 zip_safe = True
 install_requires = 
-	PyPDF2 >= 2.1.1, < 3
+ pypdf >= 3.0, < 5 
 
 [options.entry_points]
 console_scripts = 
